### PR TITLE
DG appendix section and code reuse

### LINKED ISSUE #225: Developers guide (Appendix: requirements + instructions for manual testing)
update these 2 sections

### PR DESCRIPTION
**Pull request by tengcharmaine**
Closes #225 and #257. 

I added a line of code reuse for both CoPilot and ChatGPT in the acknowledgements sections. Can go see if its okay or if yall want to add anything else.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -444,28 +446,38 @@ The following activity diagram summarizes what happens when a user executes a ne
...
-| `* `    | user | show individual patient information                                        | focus on the patients details during consultations                            |
+| Priority | As a …​                     | I want to …​                                                         | So that I can…​                                                                                       |
+|------|-----------------------------|----------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------|
+| `* * *` | user                        | add a new patient                                                    | manage patient information for that new patient                                                       |
+| `* * *` | tidy user                   | easily delete unnecessary data to reduce clutter in ClinicMate       | maintain a clean and organised patient database                                                       |
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: maybe organised user will be better?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -444,28 +446,38 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+| `* * *` | user                        | add notes associated to each patient (including past diagnosis etc.) | easily follow up on necessary actions and understand the patient better                               |
+| `* * *` | user                        | exit ClinicMate quickily                                             | conclude my session at the end of the day and ensure that ClinicMate is not running in the background |
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: quickly

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -444,28 +446,38 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+| `* *` | user                        | remove notes associated to each patient                              | remove irrelevant information and keep my notse up to data                                            |
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: notes instead of notse

Comment (Jayne1010) [reply docs/DeveloperGuide.md]: date instead of data

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -444,28 +446,38 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+| `* *` | user managing many patients | filter the data based on ic number                                   | view the information quickily without searching through the whole list                                |
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: quickly

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -444,28 +446,38 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+| `* *` | careless user               | edit patient information                                             | keep accurate records of my patients                                                                  |
+| `* *` | new user                    | see usage instructions                                               | refer to instructions when I forget how to use ClinicMate                                             |
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: when I have difficulty using ClinicMate?

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (jiahui0309) [inline docs/DeveloperGuide.md]: notes?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -444,28 +446,38 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+| `* *` | new user                    | clear all sample data                                                | fill ClinicMate with all my patients information                                                      |
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: insert my own patient information into ClinicMate?

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -444,28 +446,38 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+| `* *` | new user                    | see ClinicMate populated with sample data                            | see how ClinicMate looks like when it is in use                                                       |
+| `*`  | user                        | show individual patient information                                  | focus on the patients details during consultations                                                    |
+| `*`  | creative user               | be able to change the theme of ClinicMate                            | personalise the apperance of the user interface based on my preferences                               |
+| `*`  | new user                    | be able to export patient data for reporting purposes                | analyse health trends in a patient over time                                                          |
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: analyse health trends of a patient?

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -444,28 +446,38 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+| `* *` | careless user               | edit patient information                                             | keep accurate records of my patients                                                                  |
+| `* *` | new user                    | see usage instructions                                               | refer to instructions when I forget how to use ClinicMate                                             |
+| `* *` | new user                    | clear all sample data                                                | fill ClinicMate with all my patients information                                                      |
+| `* *` | new user                    | see ClinicMate populated with sample data                            | see how ClinicMate looks like when it is in use                                                       |
+| `*`  | user                        | show individual patient information                                  | focus on the patients details during consultations                                                    |
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: patient's details?

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -444,28 +446,38 @@ The following activity diagram summarizes what happens when a user executes a ne
...
 
 **MSS**
 
-1.  User requests to list persons
-2.  ClinicMate shows a list of persons
-3.  User requests to add a specific person in the list
-4.  ClinicMate adds the person
+1.  User requests to list patients.
+2.  ClinicMate shows a list of patients.
+3.  User requests to add a specific patient in the list.
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: add a specific patient to the list

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -562,26 +583,39 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
 **MSS**
 
-1.  User requests to list persons
-2.  ClinicMate shows a list of persons
-3.  User requests to edit a specific person in the list
-4.  ClinicMate edits the person
+1.  User requests to list patients.
+2.  ClinicMate shows a list of patients. 
+3.  User requests clear all patients from the full list of patients.
```
Comment (jiahui0309) [inline docs/DeveloperGuide.md]: requests to?

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -533,26 +541,39 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
   Use case ends.
 
-* 3a. The given IC number is invalid.
+* 3a. The given command format is wrong → handled similarly to UC01 3a.
 
-    * 3a1. ClinicMate shows an error message.
+* 4a. The given IC number is invalid → handled similarly to UC02 4a.
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: should IC_NUMBER be in code block or nah

**Review by jiahui0309 — CHANGES_REQUESTED** [review]
some minor changes!

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -884,7 +948,24 @@ testers are expected to do more *exploratory* testing.
...
        Expected: Similar to previous.
 
+### Showing a person
+
+1. Showing the note of a person while all persons are being shown
+
+    1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+
+    1. Test case: `show T0123456A`<br>
+       Expected: The note of the contact with the unique identification number `T0123456A` will be shown in the patient notes panel. Successful show message will be shown in the status message.
```
Comment (Jayne1010) [inline docs/DeveloperGuide.md]: change to IC number?

**Review by Jayne1010 — APPROVED** [review]
LGTM!

**Pull request merged**